dask/bytes/core.py

```diff
@@ -15,10 +15,57 @@
 # Global registration dictionaries for backend storage functions
 # See docstrings to functions below for more information
 _read_bytes = dict()
+_write_bytes = dict()
 _open_files = dict()
 _open_text_files = dict()
 
 
+def write_bytes(values, urlpath, compression=None, **kwargs):
+    """For a list of values which evaluate to byte, produce delayed values
+    which, when executed, result in writing to files.
+
+    The path maybe a concrete directory, in which case it is interpreted
+    as a directory, or a template for numbered output.
+
+    The path may be preceded by a protocol, like ``s3://`` or ``hdfs://`` if
+    those libraries are installed.
+
+    Parameters
+    ----------
+    urlpath: string
+        Absolute or relative filepath, URL (may include protocols like
+        ``s3://``), which may be a template (include `{}`).
+    compression: string or None
+        String like 'gzip' or 'xz'.  Must support efficient random access.
+    **kwargs: dict
+        Extra options that make sense to a particular storage connection, e.g.
+        host, port, username, password, etc.
+
+    Examples
+    --------
+    >>> values = write_bytes('s3://bucket/part-{}.csv')  # doctest: +SKIP
```

Comment:
Thoughts on using `*` instead?  It might be valuable to look at the interface used by `dask.bag.to_textfiles` or `dask.dataframe.to_csv` in name_function in order to ensure that this functionality can easily supplant the logic in both of those functions.